File: typespec/problem/admin/hub-signup-domains.ts
import type { Details } from "../details.ts";

export const HubSignupDomainNotFoundError: Readonly<Details> = {
  type: "vetchium-problem-details/hub-signup-domain-not-found",
  title: "Hub signup domain not found",
  status: 404,
  detail: "The requested Hub signup domain does not exist",
};

export const HubSignupDomainAlreadyExistsError: Readonly<Details> = {
  type: "vetchium-problem-details/hub-signup-domain-already-exists",
  title: "Hub signup domain already exists",
  status: 409,
  detail: "The normalized domain is already in the Hub signup allowlist",
};

File: typespec/problem/admin/hub_signup_domains.go
package admin

import "github.com/vetchium/src/typespec/problem"

var HubSignupDomainNotFoundError = problem.Details{
	Type:   "vetchium-problem-details/hub-signup-domain-not-found",
	Title:  "Hub signup domain not found",
	Status: 404,
	Detail: "The requested Hub signup domain does not exist",
}

var HubSignupDomainAlreadyExistsError = problem.Details{
	Type:   "vetchium-problem-details/hub-signup-domain-already-exists",
	Title:  "Hub signup domain already exists",
	Status: 409,
	Detail: "The normalized domain is already in the Hub signup allowlist",
}
